LoopKit/RemoteData.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  RemoteData.swift
+//  LoopKit
+//
+//  Created by Darin Krauss on 5/21/19.
+//  Copyright © 2019 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+
+
+public protocol RemoteData {
+
+    func uploadSettings(_ settings: Settings, lastUpdated: Date)
+
+    func uploadLoopStatus(
+        insulinOnBoard: InsulinValue?,
+        carbsOnBoard: CarbValue?,
+        predictedGlucose: [GlucoseValue]?,
+        recommendedTempBasal: (recommendation: TempBasalRecommendation, date: Date)?,
+        recommendedBolus: Double?,
+        lastReservoirValue: ReservoirValue?,
+        pumpManagerStatus: PumpManagerStatus?,
+        glucoseTargetRangeSchedule: GlucoseRangeSchedule?,
+        scheduleOverride: TemporaryScheduleOverride?,
+        glucoseTargetRangeScheduleApplyingOverrideIfActive: GlucoseRangeSchedule?,
+        loopError: Error?)
+
+    func upload(glucoseValues values: [GlucoseValue], sensorState: SensorDisplayable?)
+
+    func upload(pumpEvents events: [PersistedPumpEvent], fromSource source: String, completion: @escaping (Result<[URL], Error>) -> Void)
```

Comment:
At some point I would imagine that this (and the other datatypes) would change to be query initiated by the service to ask the delegate for events, passing a query anchor that this service manages?  What is in this PR is fine as it matches the current paradigm well. Just checking that we're on the same page for eventual implementation.

Reply:
Yes, the next chunk of work is to allow the services to "pull" the events from the delegate using anchors and such. This is really just an intermediate (but fully functional) step on the way to the final solution.